screenshots-diff/diffImagesAsync.ts

```diff
@@ -271,10 +271,18 @@ export const diffImagesAsync = async (options: {
   // tslint:enable no-bitwise
 
   if (mismatchedPixels !== 0) {
-    diff.pack().pipe(fs.createWriteStream(diffImagePath));
+    return new Promise((resolve) => {
+      const diffImageFileSteam = fs.createWriteStream(diffImagePath);
+      diff.pack().pipe(diffImageFileSteam);
+
+      // Only resolve once we're done writing the image
+      diffImageFileSteam.on("close", () => {
+        resolve({ mismatchedPixels, diffHash });
+      })
+    });
+  } else {

```

Comment:
Not sure what the async API gives us here. Maybe we should switch the sync API of PNGjs

```
fs.writeFileSync(diffImagePath, PNG.sync.write(diff));
```

Reply:
Nevermind :p